lib/broccoli/broccoli-typescript.js

```diff
@@ -147,6 +149,33 @@ var DiffingTSCompiler = (function () {
             }
         }
     };
+    /**
+     * There is a bug in TypeScript 1.6, where the sourceRoot and inlineSourceMap properties
+     * are exclusive. This means that the sources property always contains relative paths
+     * (e.g, ../../../../angular2/src/di/injector.ts).
+     *
+     * Here, we normalize the sources property and remove the ../../../
+     *
+     * This issue is fixed in https://github.com/Microsoft/TypeScript/pull/5620.
+     * Once we switch to TypeScript 1.8, we can remove this method.
+     */
+    DiffingTSCompiler.prototype.fixSourceMapSources = function (content) {
+        try {
+            var marker = "//# sourceMappingURL=data:application/json;base64,";
+            var index = content.indexOf(marker);
+            if (index == -1)
```

Comment:
I'd either move the return to the same line of the if, or include the curly braces. Also, what about declaring all the variables on top of the function to avoid confusion because of function hoisting?